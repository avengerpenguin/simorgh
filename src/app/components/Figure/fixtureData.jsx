import React from 'react';
import FigureContainer from '../../containers/Figure';

const imageAlt = 'Pauline Clayton';
const imageSrc =
  'https://ichef.bbci.co.uk/news/640/cpsprodpb/A933/production/_101651334_bouquet_pa.jpg';
const imageRatio = 56.25;
const captionValue = 'This is a caption';
const captionValueContainingLink =
  'This is a caption [with a link](https://bbc.com/news)';
const copyrightText = 'Getty Images';

const generateFixtureData = (caption, copyright) => (
  <FigureContainer
    src={imageSrc}
    alt={imageAlt}
    ratio={imageRatio}
    caption={caption}
    copyright={copyright}
  />
);

export const FigureImage = generateFixtureData();

export const FigureImageWithCaption = generateFixtureData(captionValue);

export const FigureImageWithCaptionContainingLink = generateFixtureData(
  captionValueContainingLink,
);

export const FigureImageWithCopyright = generateFixtureData(
  null,
  copyrightText,
);

export const FigureImageWithCopyrightAndCaption = generateFixtureData(
  captionValue,
  copyrightText,
);
