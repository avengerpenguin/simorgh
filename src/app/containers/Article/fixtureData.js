import { blockContainingText } from '../../models/blocks';

export const serviceContextNews = {
  brandName: 'BBC News',
  service: 'News',
};

export const serviceContextPersian = {
  brandName: 'BBC News فارسی',
  service: 'Persian',
};

const headlineBlock = headline => blockContainingText('headline', headline);

const articleDataBuilder = (
  id,
  passportLanguage,
  home,
  headlineText,
  paragraphText,
  seoHeadline,
  promoHeadline,
  summary,
) => ({
  metadata: {
    id: `urn:bbc:ares::article:${id}`,
    locators: {
      optimoUrn: `urn:bbc:optimo:asset:${id}`,
    },
    type: 'article',
    createdBy: '',
    blockId: 'a-1',
    created: 1514808060000,
    firstPublished: 1514808060000,
    lastPublished: 1514811600000,
    lastUpdated: 1514811600000,
    passport: {
      language: passportLanguage,
      home: 'http://www.bbc.co.uk/ontologies/passport/home/Persian',
      category: 'news',
      genre: null,
    },
    tags: {},
  },
  content: {
    model: {
      blocks: [
        headlineBlock(headlineText),
        {
          type: 'text',
          blockId: 't-2',
          model: {
            blocks: [
              {
                blockId: 'p-2',
                type: 'paragraph',
                model: {
                  text: paragraphText,
                },
              },
            ],
          },
        },
      ],
    },
  },
  promo: {
    id: `urn:bbc:ares::article:${id}`,
    headlines: {
      seoHeadline,
      promoHeadline,
    },
    locators: {
      optimoUrn: `urn:bbc:optimo:asset:${id}`,
    },
    summary,
    timestamp: 1514811600000,
  },
});

export const articleDataNews = articleDataBuilder(
  'c0000000001o',
  'en-gb',
  'http://www.bbc.co.uk/ontologies/passport/home/News',
  'Article Headline',
  'A paragraph.',
  'Article Headline for SEO',
  'Article Headline for Promo',
  'Article summary.',
);

export const articleDataPersian = articleDataBuilder(
  'c0000000028o',
  'fa',
  'http://www.bbc.co.uk/ontologies/passport/home/Persian',
  'سرصفحه مقاله',
  'یک پاراگراف.',
  'سرصفحه مقاله',
  'سرصفحه مقاله برای ارتقاء',
  'خلاصه مقاله',
);
