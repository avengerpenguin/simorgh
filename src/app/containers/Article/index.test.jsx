import React from 'react';
import { shouldShallowMatchSnapshot } from '../../helpers/tests/testHelpers';
import ArticleContainer from './index';
import { articleDataNews, articleDataPersian } from './fixtureData';

// explicitly ignore console.log errors for Article/index:getInitialProps() error logging
global.console.log = jest.fn();

describe('ArticleContainer', () => {
  const newsData = { data: articleDataNews, amp: false, service: 'news' };

  const persianData = {
    data: articleDataPersian,
    amp: false,
    service: 'persian',
  };

  describe('Component', () => {
    shouldShallowMatchSnapshot(
      'should render correctly',
      <ArticleContainer data={newsData} />,
    );

    shouldShallowMatchSnapshot(
      'should render Persian article correctly',
      <ArticleContainer data={persianData} />,
    );

    describe('no data', () => {
      shouldShallowMatchSnapshot(
        'should render correctly',
        <ArticleContainer />,
      );
    });
  });
});
