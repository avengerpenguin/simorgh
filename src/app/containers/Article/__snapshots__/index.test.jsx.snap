// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContainer Component no data should render correctly 1`] = `null`;

exports[`ArticleContainer Component should render Persian article correctly 1`] = `
<React.Fragment>
  <ServiceContextProvider
    service="persian"
  >
    <Header />
    <Metadata
      amp={false}
      articleAuthor="https://www.facebook.com/bbcnews"
      articleSection={null}
      canonicalLink="https://www.bbc.com/persian/articles/c0000000028o"
      defaultImage="https://news.files.bbci.co.uk/ws/img/logos/og/persian.png"
      defaultImageAltText="BBC News فارسی"
      description="خلاصه مقاله"
      lang="fa"
      locale="fa"
      metaTags={Array []}
      opengraphSiteName="BBC News فارسی"
      timeFirstPublished="2018-01-01T12:01:00.000Z"
      timeLastUpdated="2018-01-01T13:00:00.000Z"
      title="سرصفحه مقاله"
      twitterCreator="@bbcpersian"
      twitterSite="@bbcpersian"
      type="article"
    />
    <MainContentContainer
      blocks={
        Array [
          Object {
            "model": Object {
              "blocks": Array [
                Object {
                  "blockId": "t-1",
                  "model": Object {
                    "blocks": Array [
                      Object {
                        "blockId": "p-1",
                        "model": Object {
                          "text": "سرصفحه مقاله",
                        },
                        "type": "paragraph",
                      },
                    ],
                  },
                  "type": "text",
                },
              ],
            },
            "type": "headline",
          },
          Object {
            "blockId": "t-2",
            "model": Object {
              "blocks": Array [
                Object {
                  "blockId": "p-2",
                  "model": Object {
                    "text": "یک پاراگراف.",
                  },
                  "type": "paragraph",
                },
              ],
            },
            "type": "text",
          },
        ]
      }
    />
    <FooterContainer />
  </ServiceContextProvider>
</React.Fragment>
`;

exports[`ArticleContainer Component should render correctly 1`] = `
<React.Fragment>
  <ServiceContextProvider
    service="news"
  >
    <Header />
    <Metadata
      amp={false}
      articleAuthor="https://www.facebook.com/bbcnews"
      articleSection={null}
      canonicalLink="https://www.bbc.com/news/articles/c0000000001o"
      defaultImage="https://www.bbc.co.uk/news/special/2015/newsspec_10857/bbc_news_logo.png?cb=1"
      defaultImageAltText="BBC News"
      description="Article summary."
      lang="en-gb"
      locale="en_GB"
      metaTags={Array []}
      opengraphSiteName="BBC News"
      timeFirstPublished="2018-01-01T12:01:00.000Z"
      timeLastUpdated="2018-01-01T13:00:00.000Z"
      title="Article Headline for SEO"
      twitterCreator="@BBCNews"
      twitterSite="@BBCNews"
      type="article"
    />
    <MainContentContainer
      blocks={
        Array [
          Object {
            "model": Object {
              "blocks": Array [
                Object {
                  "blockId": "t-1",
                  "model": Object {
                    "blocks": Array [
                      Object {
                        "blockId": "p-1",
                        "model": Object {
                          "text": "Article Headline",
                        },
                        "type": "paragraph",
                      },
                    ],
                  },
                  "type": "text",
                },
              ],
            },
            "type": "headline",
          },
          Object {
            "blockId": "t-2",
            "model": Object {
              "blocks": Array [
                Object {
                  "blockId": "p-2",
                  "model": Object {
                    "text": "A paragraph.",
                  },
                  "type": "paragraph",
                },
              ],
            },
            "type": "text",
          },
        ]
      }
    />
    <FooterContainer />
  </ServiceContextProvider>
</React.Fragment>
`;
